### PR TITLE
MGDAPI-5152 - Changes to support OpenShift v4.13

### PR DESCRIPTION
**Pull request by adam-cattermole**
# Issue link
[MGDAPI-5152](https://issues.redhat.com/browse/MGDAPI-5152)

# What
- Update various dependencies to support installation on OpenShift v4.13 - primarily to explicitly set the GrpcPodConfig in CatalogSource CR's.
- This relies on an update to do the same in the Observability Operator, as well as a dependency update from 3scale Operator.
	- Observability Operator PR https://github.com/redhat-developer/observability-operator/pull/197
	- 3scale Operator PR https://github.com/3scale/3scale-operator/pull/805

# Outstanding
* Once the above PR's are merged I will need to update the go.mod replaces to point to the official versions of the above
* This is currently using a custom index for the observability operator with the required changes. Once the changes have been merged the new release of the operator will need to be consumed in the installation.yaml

# Verification steps

* Provision a v4.13 cluster and deploy RHOAM from this branch

```
LOCAL=false make cluster/prepare/local
LOCAL=false USE_CLUSTER_STORAGE=true make code/run
```

* Watch the RHMI CR during installation - it should complete successfully
* Verify no alerts are firing
* Trigger uninstall and watch RHMI CR - expect RHOAM to successfully uninstall
```
oc delete rhmi rhoam -n redhat-rhoam-operator
```

### CODE REVIEW

**Comment by valerymo** [comment]
Rhoam installed ok for me. Alerts firing - DeadMansSwitchAlert (known) and RHOAMThreescaleCriticalMetricsMissing (?).
But I'd like to note about errors - looks like all errors happen in the PR - because of same issue - `pkg/products/cloudresources/prometheusRules.go:84:22: DefaultPostgresAllocatedStorageMetricName not declared by package resources`.  
`DefaultPostgresAllocatedStorageMetricName` was added to Rhoam master recently, but removed from latest CRO. It exists in CRO v0.44.0, but missing in CRO v1.0.1, that used by latest Rhoam. cc @laurafitzgerald , @MStokluska , @carlkyrillos

File: go.mod [hunk 1]
```diff
@@ -26,29 +26,29 @@ require (
 	github.com/integr8ly/cloud-resource-operator v1.0.1
 	github.com/integr8ly/keycloak-client v0.1.11
 	github.com/noobaa/noobaa-operator/v5 v5.11.0
-	github.com/onsi/gomega v1.20.1
+	github.com/onsi/gomega v1.23.0
 	github.com/openshift/api v3.9.1-0.20191031084152-11eee842dafd+incompatible
-	github.com/operator-framework/api v0.17.1
+	github.com/operator-framework/api v0.17.3
 	github.com/operator-framework/operator-lifecycle-manager v0.21.1
```
Comment (valerymo) [inline go.mod:32]: @adam-cattermole , I'd like to ask - if need move to latest version of operator-lifecycle-manager - https://github.com/operator-framework/operator-lifecycle-manager/releases/tag/v0.24.0  , as you have commit to address [MGDAPI-4898]. 
Moving from operator-framework/operator-lifecycle-manager/pkg/api/apis/operators to operator-framework/api/pkg/operators looks very good. 
Just wanted to ask about operator-lifecycle-manager v0.21.1, although it used just in 2 ".go" files + hardcoded in some several MCG CSVs, and maybe more. But maybe it's part of this task. Thank you

**Comment by valerymo** [comment]
- attaching file with test details   
[test_MGDAPI-5152.txt](https://github.com/integr8ly/integreatly-operator/files/11135711/test_MGDAPI-5152.txt)

- Test Summary:
-- Install - Successfull, notes:  
	Alerts firing: DeadMansSwitchAlert , RHOAMThreescaleCriticalMetricsMissing  
-- Uninstall - Successfull, notes:
	Remains: CMs and rhoam-operator-metrics-service

**Comment by adam-cattermole** [comment]
@KevFan I think I figured the e2e will fail due to differences in my custom 3scale dev image - once new v0.11.5 is out I can adjust last commit and try again

**Comment by adam-cattermole** [comment]
Tests passing, going to push maxOpenshiftVersion change

**Comment by KevFan** [comment]
* Verified upgrade from master to this branch on 4.12 cluster - no downtime was detected

image

* Verified fresh installation on 4.13 Cluster

/lgtm

**Pull request merged**